### DISCUSSION TITLE
How to register for events on InputFields (TextField, Editor etc...)

### DISCUSSION THREAD

**Discussion by StepKie** [enisn/UraniumUI#248] [Q&A]
I want to achieve the equivalent to the following for `material:TextField` and `material:EditorField`:

```xaml
<Entry Text="Test">
    <Entry.Behaviors>
        <toolkit:EventToCommandBehavior Command="{Binding CmdToCallWhenUnfocusingEntry}" EventName="Unfocused" />
    </Entry.Behaviors>
</Entry>
```

i.e. I want to react to certain events like Focused, Unfocused, Completed etc. that exist for plain MAUI controls.
Obviously, the naive approach

```xaml
<material:TextField Title="{resx:Localize ServerURL}" Text="{Binding UrlOfServer}">
    <material:TextField.Behaviors>
        <toolkit:EventToCommandBehavior Command="{Binding CheckUrlCommand}" EventName="Unfocused" />
        <toolkit:EventToCommandBehavior Command="{Binding CheckUrlCommand}" EventName="Completed" />
    </material:TextField.Behaviors>
</material:TextField>
```

does not work (I guess since InputField is inherited from Grid?)

_Edit_: I have checked the docs and only found [validations](https://enisn-projects.io/docs/en/uranium/latest/validations/Index), which are not doing quite what I want.

So if I want to do achieve this and not rely on an InputForm validation triggered by a separate button press, how would I go about it?
I know that everything regarding Behaviors for custom controls is not trivial (attached behaviors, etc.). Is there something missing in the implementation, or did I miss anything in the docs?

_Edit 2_: On further attempts, I noticed that for the following:

- The ReturnCommand is called correctly, but this will only react to Enter on the soft keyboard, no other focus change.
- Neither the TextChanged event (which is actually defined in material:TextField) nor the Unfocused event (inherited from VisualElement) or its corresponding EventToCommandBehavior will be triggered. Maybe this is not marshalled correctly to/from the underlying EntryView?

```xaml
<material:TextField
   Title="{resx:Localize ServerURL}"
   ReturnCommand="{Binding EntryUnfocused}"
   Text="{Binding UrlOfServer}"
   TextChanged="TextField_Unfocused"
   Unfocused="TextField_Unfocused">
      <material:TextField.Behaviors>
         <toolkit:EventToCommandBehavior Command="{Binding EntryUnfocused}" EventName="Unfocused" />
      </material:TextField.Behaviors>
</material:TextField>
```

**Comment by enisn** [comment]
I'll check this, events might be not triggered

**Reply by StepKie** [reply]
> I'll check this, events might be not triggered

@enisn did you find the time to check this yet? Should an issue be created?

**Reply by StepKie** [reply]
I will go ahead and add convert this into an issue.
Edit: https://github.com/enisn/UraniumUI/issues/278
